File: test/h5cpp_tests/property_tests/chunk_cache_parameters.py
#
# (c) Copyright 2018 DESY,
#
# This file is part of python-pninexus.
#
# python-pninexus is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 2 of the License, or
# (at your option) any later version.
#
# python-pninexus is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with python-pninexus.  If not, see <http://www.gnu.org/licenses/>.
# ===========================================================================
#
# Created on: Jan 26, 2018
#     Author: Eugen Wintersberger <eugen.wintersberger@desy.de>
#
import unittest

from pninexus.h5cpp.property import ChunkCacheParameters


class ChunkCacheParametersTest(unittest.TestCase):

    def testConstruction(self):

        params = ChunkCacheParameters(1000000, 20313, 9.234)
        self.assertEqual(params.chunk_slots, 1000000)
        self.assertEqual(params.chunk_cache_size, 20313)
        self.assertEqual(params.preemption_policy, 9.234)

    def testSetters(self):

        params = ChunkCacheParameters()
        params.chunk_slots = 3400000
        self.assertEqual(params.chunk_slots, 3400000)

        params.chunk_cache_size = 20313
        self.assertEqual(params.chunk_cache_size, 20313)

        params.preemption_policy = 934.234
        self.assertEqual(params.preemption_policy, 934.234)

File: src/cpp/h5cpp/property/chunk_cache_parameters.cpp
//
// (c) Copyright 2018 DESY
//
// This file is part of python-pninexus.
//
// python-pninexus is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 2 of the License, or
// (at your option) any later version.
//
// python-pninexus is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with python-pninexus.  If not, see <http://www.gnu.org/licenses/>.
// ===========================================================================
//
// Created on: Jan 26, 2018
//     Author: Eugen Wintersberger <eugen.wintersberger@desy.de>
//
#include <boost/python.hpp>
#include <h5cpp/hdf5.hpp>
#include <h5cpp/contrib/stl/stl.hpp>

void create_chunk_cache_parameters_wrapper()
{
  using namespace boost::python;
  using namespace hdf5::property;

  size_t (ChunkCacheParameters::*get_chunk_slots)() const = &ChunkCacheParameters::chunk_slots;
  void (ChunkCacheParameters::*set_chunk_slots)(size_t) = &ChunkCacheParameters::chunk_slots;
  size_t (ChunkCacheParameters::*get_chunk_cache_size)() const = &ChunkCacheParameters::chunk_cache_size;
  void (ChunkCacheParameters::*set_chunk_cache_size)(size_t) = &ChunkCacheParameters::chunk_cache_size;
  double (ChunkCacheParameters::*get_preemption_policy)() const = &ChunkCacheParameters::preemption_policy;
  void (ChunkCacheParameters::*set_preemption_policy)(double) = &ChunkCacheParameters::preemption_policy;
  class_<ChunkCacheParameters>("ChunkCacheParameters")
      .def(init<>())
      .def(init<size_t,size_t,double>())
      .add_property("chunk_slots",get_chunk_slots,set_chunk_slots)
      .add_property("chunk_cache_size",get_chunk_cache_size,set_chunk_cache_size)
      .add_property("preemption_policy",get_preemption_policy,set_preemption_policy)
      ;
}
